modular_skyrat/modules/new_meteors/code/new_meteors.dm

```diff
@@ -0,0 +1,69 @@
+// 1 tick = 2 seconds
+#define METEOR_WAVE_MIN_NOTICE 210 // 7 minutes (420 seconds)
+#define METEOR_WAVE_MAX_NOTICE 270 // 9 minutes (540 seconds)
+#define METEOR_WAVE_DURATION 37 // 74 seconds
+
+/datum/round_event_control/meteor_wave
+	///Where will the meteors be coming from? -- Established in admin_setup, passed down to round_event
+	var/start_side
+
+/datum/round_event/meteor_wave
+	///Which direction the storm will come from.
+	var/start_side
+
+/datum/round_event/meteor_wave/New()
+	..()
+	start_when = rand(METEOR_WAVE_MIN_NOTICE, METEOR_WAVE_MAX_NOTICE)
+	end_when = start_when + METEOR_WAVE_DURATION
+
+/datum/round_event_control/meteor_wave/admin_setup()
+	if(!check_rights(R_FUN))
+		return ADMIN_CANCEL_EVENT
+
+	var/force_dir = tgui_alert(usr, "Choose a side to throw rocks at?", "Interrupt some ERP?", list("Yes", "No", "Cancel"))
+	if(force_dir == "Cancel")
+		return ADMIN_CANCEL_EVENT
+
+	if(force_dir == "Yes")
+		var/chosen_direction = tgui_input_list(usr, "Pick one!","Plausible Deniability.", list("North", "South", "East", "West"))
+		switch(chosen_direction)
+			if("North")
+				start_side = NORTH
+			if("South")
+				start_side = SOUTH
+			if("East")
+				start_side = EAST
+			if("West")
+				start_side = WEST
```

Comment:
Early returns are nice and simple, and this way you also properly cover the event that admins click the "X" button instead.
```suggestion
	if(force_dir != "Yes")
		return ADMIN_CANCEL_EVENT

	var/chosen_direction = tgui_input_list(usr, "Pick one!","Plausible Deniability.", list("North", "South", "East", "West"))
	switch(chosen_direction)
		if("North")
			start_side = NORTH
		if("South")
			start_side = SOUTH
		if("East")
			start_side = EAST
		if("West")
			start_side = WEST
```